docs/.pre-commit-config.yaml

```
@@ -0,0 +1,46 @@
+---
+repos:
+  # import formatter with black configurations
+    - repo: https://github.com/pycqa/isort
+      rev: 5.10.1
+      hooks:
+          - id: isort
+            name: isort (python)
+            args: [--profile, black, --filter-files]
+
+  # Code formatter for both python files and jupyter notebooks
+    - repo: https://github.com/psf/black
+      rev: 22.10.0
+      hooks:
+          - id: black-jupyter
+          - id: black
+            language_version: python3.10
+
+    # snakemake formatting
+    - repo: https://github.com/snakemake/snakefmt
+      rev: v0.8.0
+      hooks:
+          - id: snakefmt
+
+    # fast linter written in rust
+    - repo: https://github.com/charliermarsh/ruff-pre-commit
+    # Ruff version.
+      rev: v0.0.218
```

Consider revising the comment to match the indent of the line below.
```suggestion
      # Ruff version.
      rev: v0.0.218
```